### DISCUSSION TITLE
Create virtual in typegoose

### DISCUSSION THREAD

**Discussion by vasyl97kharchenko** [typegoose/typegoose#900] [Q&A]
Hello, I need your help. How do I transform this in mongoose to typegoose so that when I do entity.populate('subtotal') -> I get subtotal populated value?
```js
schema.virtual('subtotal').get(function subtotal() {
    return this.total - this.tax - this.shipping;
});
```
Is it the only way to do so?
```js
class Entity {
  ...
  get subtotal(): number {
      return this.total - this.tax - this.shipping;
    }
}
```

**Comment by hasezoey** [comment]
> Is it the only way to do so?

yes, class `get & set` directly map to mongoose's `virtual.get & set`

there is also the [Virtuals](https://typegoose.github.io/typegoose/docs/api/virtuals/) documentation

---

aside from that `entity.populate('subtotal')` does not look valid to me (`subtotal` is not a populate-able path in both examples)
